client/src/main/java/io/getstream/chat/android/client/utils/Result.kt

```diff
@@ -2,27 +2,25 @@ package io.getstream.chat.android.client.utils
 
 import io.getstream.chat.android.client.errors.ChatError
 
-data class Result<T>(
+public data class Result<T : Any>(
```

Comment:
Since this class is used as a read-only class by clients, and they never need to instantiate it, should we just hide it behind an interface?

Reply:
This class needs to be refactored to have something like [Either](https://arrow-kt.io/docs/apidocs/arrow-core-data/arrow.core/-either/), that at the end is the motivation of this class.
The problem is that it is "a core class" in our SDK, for now we can keep it as it is